### DISCUSSION TITLE
Combine the start of a GCode with the start of the spindle

### DISCUSSION THREAD

**Discussion by dasarne** [winder/Universal-G-Code-Sender#2091] [Help]
Hi,
the path workbench of Freecad does not write a spindle-on (M3/M4) at the start of the GCode for grbl. Maybe this is not a bad idea, because at the stop/start of a GCode the spindle handling has to be considered as well.
So far, I have always started a macro before executing my GCode to switch on the spindle, but this can be forgotten.
So here is my question: 
Is it possible to combine the start of the execution of GCode with the switch on of the spindle?

Switching off at the end of the program and at stop works perfectly reliable.

Yours
 Arne

**Comment by chrisla23** [comment]
It looks like you can customize the post-processor of Freecad to do that:

https://wiki.freecadweb.org/Path_Postprocessor_Customization

Most call CAD/CAM software has the ability to modify a header to be
inserted before every gcode.

-Chris

**Reply by dasarne** [reply]
Thank you for the answer. I didn't know about that, and it will at least make it easier to start a GCode.
But wouldn't it be great if I could link one macro to the start of a GCode and another to the stop? For example, one could go to the Safe Hight first when Stop and then turn off the spindle. On start, the spindle could be switched on. 
I would also think of a macro for the tool change (M6) or what should be executed in case of an error.
What do you think of that idea?
